### PR TITLE
Implement LPOracle Factory

### LINKED ISSUE #18: Oracle Factory Contract
Use create2 for deterministic oracles. Here is a sketch interface:

```ts
interface ILPOracleFactory {
    function deployOracle(address pool, address feed0, address feed1) external returns (address oracle);
    function computeOracleAddress(address pool, address feed0, address feed1) external view returns (address);
    function getOracle(address pool) external view returns (address);
}
```

Something to consider:
⚠️ Deterministic addresses may make it impossible to deploy a new oracle in the event that the oracle is "Bricked" by the issue described in #15 ...

### PR DESCRIPTION
**Pull request by bh2smith**
Introducing 
- ILPOracleFactory
- LPOracleFactory
- Tests/Benchmarks

Closes #18

### CODE REVIEW

File: src/LPOracleFactory.sol [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    /// @param feed0 Chainlink USD price feed for pool token at index 0
+    /// @param feed1 Chainlink USD price feed for pool token at index 1
+    /// @return oracle Address of the newly deployed oracle
+    function deployOracle(address pool, address feed0, address feed1) external returns (address oracle) {
+        if (getOracle[pool] != address(0)) revert OracleAlreadyExists();
+        (bytes32 salt, bytes memory bytecode) = _computeDeploymentData(pool, feed0, feed1);
+
+        assembly {
+            oracle := create2(0, add(bytecode, 32), mload(bytecode), salt)
+        }
```
Comment (bh2smith) [inline src/LPOracleFactory.sol]: This gives a warning about using assembly!

Alternative suggestion
```suggestion
        // Use Solidity's create2 for deployment
        oracle = address(new LPOracle{ salt: salt }(pool, HELPER, feed0, feed1));
```

File: src/interfaces/IOracleFactory.sol [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.25;
+
+interface ILPOracleFactory {
```
Comment (bh2smith) [inline src/interfaces/IOracleFactory.sol:4]: We may not need this... what do you think @lumoswiz

File: test/unit/GetFeedData.t.sol [hunk 3]
```diff
@@ -98,5 +98,7 @@ contract GetFeedData_Unit_Test is BaseTest {
         (uint256 price0, uint256 price1, uint256 updatedAt) = oracle.exposed_getFeedData();
 
         assertEq(int256(price0), answer0);
+        assertEq(int256(price1), answer1);
+        assertEq(updatedAt, block.timestamp, "updatedAt");
```
Comment (bh2smith) [inline test/unit/GetFeedData.t.sol:102]: These were flagged as unused by compiler so I added tests here.

File: src/LPOracleFactory.sol [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import { ILPOracleFactory } from "./interfaces/IOracleFactory.sol";
+
+/// @title LPOracle Factory
+/// @notice Factory contract for deploying LPOracle instances with deterministic addresses
+contract LPOracleFactory is Ownable, ILPOracleFactory {
+    error OracleAlreadyExists();
+    error DeployFailed();
+
+    /// @notice BCoWHelper contract address used for all oracle deployments
+
```
Comment (lumoswiz) [inline src/LPOracleFactory.sol:15]: non-blocking: remove this line so natspec sits above storage variable.

File: src/interfaces/IOracleFactory.sol (hunk 2 above)
Comment (lumoswiz) [reply src/interfaces/IOracleFactory.sol:4]: Yeah, probably don't require it. We can see how it goes and remove later.

**Review by lumoswiz — APPROVED** [review]

File: src/interfaces/IOracleFactory.sol (hunk 2 above)
Comment (bh2smith) [reply src/interfaces/IOracleFactory.sol:4]: It was necessary when I had four different factories for the bench marking utility function.

File: src/LPOracleFactory.sol (hunk 4 above)
Comment (bh2smith) [reply src/LPOracleFactory.sol:15]: Oh I had a local commit with this fixed...

**Pull request merged**